src/box/tuple.h

```diff
@@ -334,19 +334,26 @@ struct PACKED tuple
 	/** Format identifier. */
 	uint16_t format_id;
 	/**
-	 * Length of the MessagePack data in raw part of the tuple.
-	 * Must be set in tuple_init_default and accessed by tuple_bsize etc.
+	 * A pair of fields for storing data offset and data bsize:
+	 * Data offset - to the MessagePack from the begin of the tuple.
+	 * Data bsize - length of the MessagePack data in raw part of the tuple.
+	 * These two values can be stored in bulky and compact modes.
+	 * In bulky mode offset is simple stored in data_offset_bsize_raw
+	 * member while bsize is stored in bsize_bulky.
+	 * In compact mode bsize_bulky is not used and both values are combined
+	 * in data_offset_bsize_raw: offset in high byte, bsize in low byte,
+	 * and the highest bit is set to 1 as a flag of compact mode.
```

Comment:
Why not use a bit flag instead? We plenty of space available for bit flags in struct tuple now.

Reply:
By design in memtx all flags are overwritten when a tuple is placed to delayed deletion list. So we can't store anything significant in lower 4 bytes of tuple.